### PR TITLE
feat(dashboard): build pay-span regex from currencyTokens list

### PR DESCRIPTION
**Pull request by wasvlad**
## What does this PR do?

Refactors the hard-coded pay-span regex in derive.go into a builder driven by an explicit currencyTokens slice, so adding a currency is a one-line append. Adds PLN, UAH, zł, ¥, ₹, ₺, ₩, ₴, JPY, CNY, INR, BRL, SEK, NOK, DKK, TRY, KRW, AUD, CAD, MXN, SGD, HKD, ZAR plus tests for the builder, suffix-no-trailing-space, and isBareSymbol.

## Type of change

- New feature — adding 23 new currencies (PLN, UAH, zł, ¥, ₹, ₺, ₩, ₴, JPY, CNY, INR, BRL, SEK, NOK, DKK, TRY, KRW, AUD, CAD, MXN, SGD, HKD, ZAR)
- Refactor (no behavior change) — for the existing currencies the matched output is identical; the regex is assembled differently

## Checklist

- [Y] I have read [CONTRIBUTING.md](https://github.com/santifer/career-ops/blob/main/CONTRIBUTING.md)
- [Y] If this is a new feature or architecture change, I opened an issue first (bug fixes, providers, docs & translations are exempt — send those straight in)
- [Y] My PR does not include personal data (CV, email, real names, scan results, or pipeline data)
- [Y] I ran `node test-all.mjs` and all tests pass
- [Y] My changes respect the [Data Contract](https://github.com/santifer/career-ops/blob/main/DATA_CONTRACT.md) (no modifications to user-layer files)
- [Y] My changes align with the [project roadmap](https://github.com/santifer/career-ops/discussions/156)

---

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved pay-amount and pay-range detection with expanded currency handling, including PLN (zł), CHF, EUR, and UAH.
  * Enhanced parsing for currency symbols and codes appearing before or after amounts.
  * Fixed suffix parsing so currency tokens don’t capture trailing spaces.
* **Tests**
  * Expanded automated coverage for additional formatting variations, mixed currency tokens, suffix edge cases, and regression scenarios.

### CODE REVIEW

File: dashboard/internal/data/derive.go [hunk 1]
```diff
@@ -36,6 +37,59 @@ var (
...
+		`~?(?:(?:%s)\s*\d[\d,]*(?:\.\d+)?[KkMm]?`+
+			`(?:\s*[-–]\s*(?:%s)?\d[\d,]*(?:\.\d+)?[KkMm]?)?`+
+			`|\d[\d,]*(?:\.\d+)?[KkMm]?`+
+			`(?:\s*[-–]\s*\d[\d,]*(?:\.\d+)?[KkMm]?)?`+
+			`\s+(?:%s))`,
+		strings.Join(prefixParts, "|"),
+		strings.Join(rangePrefixParts, "|"),
+		strings.Join(suffixParts, "|"),
+	)
+	return regexp.MustCompile(pattern)
```
[the remark this answers is not in the document]
Comment (wasvlad) [reply dashboard/internal/data/derive.go:82]: Fixed in the commit below

**Comment by wasvlad** [comment]
PR #2139 changes the same regex I touched. From my side there is no problem resolving conflict later if it lands first, it ~2 symbols change.

**Comment by santifer** [comment]
Welcome, @wasvlad. Three details made this an easy yes: `regexp.QuoteMeta` (a raw `$` in a pattern is an anchor, and that bug is invisible until someone's salary silently stops parsing), handling the suffix position so "150-200K PLN" works and not just "$150K", and returning a matches-nothing regex for an empty list instead of an empty alternation that would match everything. `go test ./...` green in the dashboard. Adding a currency is now a one-line append, which is the actual win here. Merged 🚀
